pyomo/contrib/trustregion/PyomoInterface.py

```diff
@@ -48,12 +60,15 @@ def visit(self, node, values):
         # things later; i.e., accessing the private _fcn attribute
         # below).
         for arg in list(values)[1:]:
-            if type(arg) in nonpyomo_leaf_types or arg.is_fixed():
-                # We currently do not allow constants or parameters for
+            
+            sys.stdout.write('    %s:  node is a PythonCallbackFuction\n' % arg)
+            if type(arg) in nonpyomo_leaf_types:
```

Comment:
You should treat `nonpyomo_leaf_nodes` as if they were fixed variables.  These are any leaf nodes (values) in an expression tree that are not Pyomo NumericValue types -- to it includes things like literal constants.